ov/interconnect.go

```diff
@@ -0,0 +1,228 @@
+package ov
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/HewlettPackard/oneview-golang/rest"
+	"github.com/HewlettPackard/oneview-golang/utils"
+	"github.com/docker/machine/libmachine/log"
+)
+
+type Interconnect struct {
+	Category                      string               `json:"category,omitempty"`                      // "category": "interconnects",
+	Created                       string               `json:"created,omitempty"`                       // "created": "2018-08-02T15:49:59.963Z",
+	Description                   string               `json:"description,omitempty"`                   // "description": null,
+	DeviceResetState              string               `json:"deviceResetState,omitempty"`              // "deviceResetState": "Normal",
+	EdgeVirtualBridgingAvailable  bool                 `json:"edgeVirtualBridgingAvailable,omitempty"`  // "edgeVirtualBridgingAvailable": false,
+	EnableCutThrough              bool                 `json:"enableCutThrough,omitempty"`              // "enableCutThrough": false,
+	EnableFastMacCacheFailover    bool                 `json:"enableFastMacCacheFailover,omitempty"`    // "enableFastMacCacheFailover": true,
+	EnableIgmpSnooping            bool                 `json:"enableIgmpSnooping,omitempty"`            // "enableIgmpSnooping": false,
+	EnableNetworkLoopProtection   bool                 `json:"enableNetworkLoopProtection,omitempty"`   // "enableNetworkLoopProtection": true,
+	EnablePauseFloodProtection    bool                 `json:"enablePauseFloodProtection,omitempty"`    // "enablePauseFloodProtection": true,
+	EnableRichTLV                 bool                 `json:"enableRichTLV,omitempty"`                 // "enableRichTLV": false,
+	EnableStormControl            bool                 `json:"enableStormControl,omitempty"`            // "enableStormControl": false,
+	EnableTaggedLldp              bool                 `json:"enableTaggedLldp,omitempty"`              // "enableTaggedLldp": false,
+	EnclosureName                 string               `json:"enclosureName,omitempty"`                 // "enclosureName": "SYN03_Frame1",
+	EnclosureType                 string               `json:"enclosureType,omitempty"`                 // "enclosureType": "SY12000",
+	EnclosureUri                  utils.Nstring        `json:"enclosureUri,omitempty"`                  // "enclosureUri": "/rest/enclosures/013645CN759000AC",
+	ETag                          string               `json:"eTag,omitempty"`                          // "eTag": "463bd328-ffc8-40ae-9603-6136fa9e6e58",
+	FirmwareVersion               string               `json:"firmwareVersion,omitempty"`               // "firmwareVersion": "1.3.0.1005",
+	HostName                      string               `json:"hostName,omitempty"`                      // "hostName": "VC4040F8-2TV5451754",
```

Comment:
@jyothis-gireesan - can you please check if we have all the attributes like icmLicenses, initialScopeUris ar missing here